CentralHub.Api/Services/LocalizationTargetService.cs

```diff
@@ -0,0 +1,34 @@
+using CentralHub.Api.Model;
+
+namespace CentralHub.Api.Services;
+
+public class LocalizationTargetService : ILocalizationTargetService
+{
+    private static Dictionary<int, List<Measurement>> Measurements = new Dictionary<int, List<Measurement>>();
+
+    public void AddMeasurementsAsync(int id, List<Measurement> measurements, CancellationToken token)
+    {
+        lock (Measurements)
+        {
+            if (Measurements.ContainsKey(id))
+            {
+                Measurements[id].AddRange(measurements);
+            }
+            else
+            {
+                Measurements.Add(id, measurements);
+            }
+        }
+    }
+
+    public List<Measurement> GetMeasurementsForId(int id, CancellationToken token)
+    {
+        List<Measurement> measurements;
+        lock (Measurements)
+        {
+            measurements = Measurements.GetValueOrDefault(id, new List<Measurement>());
```

Comment:
You are returning a reference to the list in the dictionary.
This is also not threadsafe. Make a copy of the list and return that